packages/ui/src/v2/types/components.ts

```diff
@@ -0,0 +1,126 @@
+/**
+ * Component Type Definitions
+ *
+ * This file exports all component interfaces and types
+ * for better IDE support and type safety.
+ */
+
```

Comment:
Are these types used? I'd expect e.g `CTButton` types to live in `ct-button/ct-button.ts`, but not sure if events/types here are referenced